plugin/pkg/auth/authenticator/token/oidc/oidc.go

```diff
@@ -238,7 +238,23 @@ func (a *OIDCAuthenticator) AuthenticateToken(value string) (user.Info, bool, er
 	var username string
 	switch a.usernameClaim {
 	case "email":
-		// TODO(yifan): Check 'email_verified' to make sure the email is valid.
+		verified, ok := claims["email_verified"]
+		if !ok {
+			return nil, false, errors.New("'email_verified' claim not present")
```

Comment:
I prefer this, but if @erictune  and @liggitt support having yet another OIDC flag for `allow-missing-email-verification`, I'm ok with that instead.  Unverified emails could do naughty things.